lab/src/Microsoft.ServiceModel.Syndication/src/Atom10FeedFormatter.cs

```diff
@@ -243,162 +223,281 @@ internal static void ReadCategory(XmlReader reader, SyndicationCategory category
-            if (reader.IsStartElement(Atom10Constants.AuthorTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Authors.Add(ReadPersonFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.CategoryTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Categories.Add(ReadCategoryFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.ContentTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Content = ReadContentFrom(reader, result);
-            }
-            else if (reader.IsStartElement(Atom10Constants.ContributorTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Contributors.Add(ReadPersonFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.IdTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Id = reader.ReadElementString();
-            }
-            else if (reader.IsStartElement(Atom10Constants.LinkTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Links.Add(ReadLinkFrom(reader, result));
-            }
-            else if (reader.IsStartElement(Atom10Constants.PublishedTag, Atom10Constants.Atom10Namespace))
-            {
-                reader.ReadStartElement();
-                result.PublishDate = DateFromString(reader.ReadString(), reader);
-                reader.ReadEndElement();
-            }
-            else if (reader.IsStartElement(Atom10Constants.RightsTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Copyright = ReadTextContentFrom(reader, "//atom:feed/atom:entry/atom:rights[@type]");
-            }
-            else if (reader.IsStartElement(Atom10Constants.SourceFeedTag, Atom10Constants.Atom10Namespace))
-            {
-                reader.ReadStartElement();
-                result.SourceFeed = ReadFeedFrom(reader, new SyndicationFeed(), true); //  isSourceFeed 
-                reader.ReadEndElement();
-            }
-            else if (reader.IsStartElement(Atom10Constants.SummaryTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Summary = ReadTextContentFrom(reader, "//atom:feed/atom:entry/atom:summary[@type]");
-            }
-            else if (reader.IsStartElement(Atom10Constants.TitleTag, Atom10Constants.Atom10Namespace))
-            {
-                result.Title = ReadTextContentFrom(reader, "//atom:feed/atom:entry/atom:title[@type]");
-            }
-            else if (reader.IsStartElement(Atom10Constants.UpdatedTag, Atom10Constants.Atom10Namespace))
-            {
-                reader.ReadStartElement();
-                result.LastUpdatedTime = DateFromString(reader.ReadString(), reader);
-                reader.ReadEndElement();
-            }
-            else
+
+            string name = reader.LocalName;
+            string ns = reader.NamespaceURI;
```

Comment:
Same thing applies here as it did to TryParseFeedElementFromAsync, need to call MoveToContentAsync and check the node type.